src/analyses/threadJoins.ml

```diff
@@ -27,8 +27,12 @@ struct
   let special ctx (lval: lval option) (f:varinfo) (arglist:exp list) : D.t =
     match LibraryFunctions.classify f.vname arglist with
     | `ThreadJoin (id, ret_var) ->
-      (
-        let threads = TIDs.elements (ctx.ask (Queries.EvalThread id)) in
+      let threads = ctx.ask (Queries.EvalThread id) in
+      if TIDs.is_top threads then
+        ctx.local (* TODO: why needed? *)
+      else (
+        (* elements throws if the thread set is top *)
```

Comment:
Does this still happen after #388?

Reply:
Yes, e.g. in `04/58` it does when I enabled this analysis. Seen that `ctx.local` is always sound here, I'd suggest to keep it such that this analysis doesn't cause exceptions in strange setting such as `04/58` or some other weirdness that one might encounter for thread ids.